### ISSUE TITLE
[Android] By default enable the "allow links to open in apps outside of Brave" setting

### ISSUE THREAD

**Issue by timchilds** [brave/brave-browser#32500]
## Description
The `allow links to open in apps outside of Brave` setting is current disabled by default on Android (app links will not open outside Brave).

This is causing some user confusion/issues, see slack thread here:
https://bravesoftware.slack.com/archives/C0816QH3R/p1692901085393929

We should change the default to enabled to match the Brave iOS app. At least until we create app-level options.

**Comment by kjozwiak** [comment]
@Uni-verse @vlad-hoff STR/Cases can be found via https://github.com/brave/brave-core/pull/19875#issuecomment-1696833181 👍 I would recommend adding a few other app links just in case. I basically just used GPS for a quick check on Nighty before uplifting into `1.58.x` & `1.57.x`.

**Comment by kjozwiak** [comment]
The above requires `1.57.56` or higher for `1.57.x` verification 👍

**Comment by kjozwiak** [comment]
@Uni-verse @vlad-hoff just a heads up that the upgrade case is not working as expected until https://github.com/brave/brave-browser/issues/32586 is fixed/uplifted into `1.57.x`.

**Comment by Uni-verse** [comment]
Verified on `Samsung Galaxy S21 5G` using version:

```
Brave	1.57.57 Chromium: 116.0.5845.163 (Official Build) (64-bit) 
Revision	d85db1f5df3b20ffecf96ab3f0dc7fca1d536955
OS	Android 13; Build/TP1A.220624.014; 33; REL
```
- Verified test cases under `Clean Install/New Profile` in https://github.com/brave/brave-core/pull/19875#issuecomment-1696833181

first launch | search results | opened in external app | disable feature | opened in brave | Waze | Waze (feature disabled)
-|-|-|-|-|-|-
![Screenshot 2023-08-30 at 10 55 31 AM](https://github.com/brave/brave-browser/assets/17885425/2aa91425-480e-48ab-bf62-af6ee4561ad9) | ![Screenshot 2023-08-30 at 10 56 20 AM](https://github.com/brave/brave-browser/assets/17885425/5ddb7090-db74-45f9-bbbc-80d3d5358b6d) | ![Screenshot 2023-08-30 at 10 56 34 AM](https://github.com/brave/brave-browser/assets/17885425/1a275509-5ed5-434c-9a33-a5d64a61a25b) | ![Screenshot 2023-08-30 at 10 57 38 AM](https://github.com/brave/brave-browser/assets/17885425/aa6fb38a-c10a-4268-94ff-2626c0555bcd) | ![Screenshot 2023-08-30 at 10 57 03 AM](https://github.com/brave/brave-browser/assets/17885425/8216c03e-3997-467a-9a13-21c3f10f9044) | ![Screenshot 2023-08-30 at 11 44 59 AM](https://github.com/brave/brave-browser/assets/17885425/4e4e6cdb-a543-4027-bb6f-aab7674143fe) | ![Screenshot 2023-08-30 at 11 45 43 AM](https://github.com/brave/brave-browser/assets/17885425/7a9113dd-a616-41f6-b19e-2753efaeb4f2)
